### PR TITLE
feat: add stop orders network parameter

### LINKED ISSUE #8516: Stop orders network parameters
A network parameter will control the maximum number of stop orders per party (suggested initial value: between 4 and 10).

### PR DESCRIPTION
**Pull request by jeremyletang**
close #8516

### CODE REVIEW

File: core/types/stop_orders.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ const (
 	// Stopped by the network, e.g: OCO other side has been triggered.
 	StopOrderStatusStopped = vega.StopOrder_STATUS_STOPPED
 	// Stop order has been triggered and generated an order.
-	StopOrderStatusTiggered = vega.StopOrder_STATUS_TRIGGERRED
+	StopOrderStatusTiggered = vega.StopOrder_STATUS_TRIGGERED
```
Comment (candida-d) [inline core/types/stop_orders.go:48]:
```suggestion
	StopOrderStatusTriggered = vega.StopOrder_STATUS_TRIGGERED
```

File: core/types/stop_orders.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ const (
...
 	// Stop order has been expired.
```
Comment (candida-d) [inline core/types/stop_orders.go:49]:
```suggestion
	// Stop order has expired.
```

**Review by Jiajia-Cui — DISMISSED** [review]
LGTM

File: core/netparams/keys.go [hunk 3]
```diff
@@ -35,6 +35,7 @@ const (
 	MarketLiquidityProvisionShapesMaxSize           = "market.liquidityProvision.shapes.maxSize"
 	MarketMinLpStakeQuantumMultiple                 = "market.liquidityProvision.minLpStakeQuantumMultiple"
 	MarketSuccessorLaunchWindow                     = "market.liquidity.successorLaunchWindowLength"
+	MarketPartiesMaximumStopOrders                  = "market.parties.maximumStopOrders"
```
[the remark this answers is not in the document]
Comment (jeremyletang) [reply core/netparams/keys.go:38]: fair enough

**Review by wwestgarth — APPROVED** [review]

**Review by candida-d — APPROVED** [review]

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**